### PR TITLE
plat-stm32mp1: remove shared_resources driver

### PR DESCRIPTION
**Pull request by etienne-lms**
Final step to remove *plat-stm32mp1/shared_resources.c* driver that has been replaced with the use of the firewall framework.

### CODE REVIEW

**Comment by GseoC** [comment]
One comment that may or may not be addressed in this P-R, then:
`Acked-by: Gatien Chevallier <gatien.chevallier@foss.st.com>`

File: core/arch/arm/plat-stm32mp1/stm32_util.h [hunk 1]
```diff
@@ -126,51 +126,6 @@ static inline int decr_refcnt(unsigned int *refcnt)
 	return decr_shrefcnt(refcnt, true);
```
[the remark this answers is not in the document]
Comment (etienne-lms) [reply core/arch/arm/plat-stm32mp1/stm32_util.h:124]: Indeed you're right, thanks. I'll append a commit to remove these.

**Comment by etienne-lms** [comment]
I've applied your review tag on the reviewed commits and appended a commit to remove the deprecated platform specific refcount helper functions.

**Comment by GseoC** [comment]
`Acked-by: Gatien Chevallier <gatien.chevallier@foss.st.com>`

**Comment by etienne-lms** [comment]
Review tag applied on all commits.

**Pull request merged**